### PR TITLE
chore(layers): moved `layer-publisher` into npm workspace

### PR DESCRIPTION
**Pull request by dreamorosi**
## Description of your changes

This PR introduces changes to the files related to publishing public layers that are aimed at:
- Moving the utility into the project's npm workspace
- Reducing technical debt on this area

The first point has already been explained at length in the linked issue (#1226), but the TL;DR; is that by bringing this code into the workspace we can make sure all shared dependencies (which happen to be almost all of them) are version aligned. This is especially desirable given that this code is CDK-heavy and that library infamously doesn't play well with version mismatches.

In regards to the second point, the following high level changes have been made:
- Renamed folder from `layer-publisher` to `layers` (same as Powertools for Python)
- Removed layers from pre-commit / pre-push hooks, all tests are done on the PR. Maintainers can still run the tests locally as needed albeit manually.
- Changed unit test structure to align more closely with the rest of the project. Also changed testing strategy moving away from snapshot testing and towards making assertions on the resources present in the generated stack.
- Changed integration test structure to align more closely with the rest of the project. Also broken down the various test cases to allow maintainers to quickly narrow down the area of code in case of issues.
- Simplified the layer publisher CDK stack by removing all logic related to spawning child processes (now done directly in the test runner)

### How to verify this change

There are three areas that would allow us to definitively say the changes are correct, one of which we can't test until after merging this PR.

The first one is by looking at the checks under this PR. The tests running as a part of this PR include unit tests of the changes. The second one are integration tests, which create a layer similar to how it would be done once pushing to prod and then tests that the file bundled in it are compatible with an actual Lambda function, [here's a successful passing run](https://github.com/awslabs/aws-lambda-powertools-typescript/actions/runs/4203851247).

The third one would be to actually publish the layers to the production AWS environment. Doing that however would cause the layer version number to increment, which is not desirable. For this reason we have opted to defer this test to the next release date, since the two previous tests already make us confident that any potential issue will be limited to the workflows in charge of publishing the layer, rather than the layer itself.

### Related issues, RFCs

**Issue number:** #1226

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [ ] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [ ] I have made corresponding changes to the *documentation*
- [ ] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [ ] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [ ] Any *dependent changes have been merged and published*
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: layers/tests/e2e/layerPublisher.test.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      functionEntry: path.join(__dirname, lambdaFunctionCodeFile),
+      tracing: Tracing.ACTIVE,
+      environment: {
+        UUID: uuid,
+        POWERTOOLS_PACKAGE_VERSION: powerToolsPackageVersion,
+        POWERTOOLS_SERVICE_NAME: 'LayerPublisherStack',
+      },
+      runtime: runtime,
+      bundling: {
+        externalModules: [
```
Comment (am29d) [inline layers/tests/e2e/layerPublisher.test.ts:84]: I am not sure why we need to bundle the modules, we already have the layer, right?

**Review by am29d — CHANGES_REQUESTED** [review]
Looks great! Two small changes:

Need to fix the path to the new file here to `bin/layers.ts` 

https://github.com/awslabs/aws-lambda-powertools-typescript/blob/1e36e45800119810d536a76427f86c24f9a3df35/layers/cdk.json#L2

File: layers/tests/e2e/layerPublisher.test.ts (hunk 1 above)
Comment (dreamorosi) [reply layers/tests/e2e/layerPublisher.test.ts:84]: Counterintuitively enough, this [means](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda_nodejs.BundlingOptions.html#externalmodules):
> A list of modules that should be considered as externals (already available in the runtime).

Given they are already in the layer like you said, we don't want `esbuild` to include them in the bundle.

**Review by am29d — APPROVED** [review]

File: layers/tests/e2e/layerPublisher.test.ts (hunk 1 above)
Comment (am29d) [reply layers/tests/e2e/layerPublisher.test.ts:84]: oh, it's the other way around 😅 

> A list of modules that should be considered as externals (already available in the runtime).

**Pull request merged**